### PR TITLE
Benji/ios 593 update emotions layout

### PR DESCRIPTION
**Pull request by Benjamindodgson**

### CODE REVIEW

File: iOS/Flows/Conversation/Emotions/EmotionContentCell.swift [hunk 1]
```diff
@@ -7,14 +7,68 @@
 //
 
 import Foundation
+import Localization
+
+struct EmotionContentModel: Hashable {
+    var emotion: Emotion?
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Emotions/EmotionContentCell.swift:13]: Helps with add an empty cell. The generic can't be nil

File: iOS/Flows/Conversation/Emotions/EmotionsCoordinator.swift [hunk 2]
```diff
@@ -20,13 +20,7 @@ class EmotionsCoordinator: PresentableCoordinator<[Emotion]> {
...
         
         self.emotionsVC.button.didSelect { [unowned self] in
-            let emotions: [Emotion] = self.emotionsVC.selectedItems.compactMap { type in
-                switch type {
-                case .emotion(let emotion):
-                    return emotion
-                }
-            }
-            self.finishFlow(with: emotions)
+            self.finishFlow(with: self.emotionsVC.selectedEmotions)
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Emotions/EmotionsCoordinator.swift:23]: the coordinator will return all the selected emotions.

File: iOS/Flows/Conversation/Emotions/EmotionsViewController.swift [hunk 3]
```diff
@@ -85,35 +76,94 @@ class EmotionsViewController: DiffableCollectionViewController<EmotionCategory,
...
-        collectionView.expandToSuperviewWidth()
-        collectionView.height = self.view.height * 0.6
-        collectionView.pin(.bottom)
+    /// The currently running task that is loading conversations.
+    private var loadTask: Task<Void, Never>?
+    
+    private func removeLastEmotion() {
+        self.loadTask?.cancel()
+        
+        self.selectedEmotions.removeFirst()
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Emotions/EmotionsViewController.swift:80]: We insert it at the beginning which is why we remove first

File: iOS/Flows/Conversation/Emotions/EmotionsViewController.swift [hunk 4]
```diff
@@ -85,35 +76,94 @@ class EmotionsViewController: DiffableCollectionViewController<EmotionCategory,
...
+        self.loadTask = Task { [weak self] in
+            guard let `self` = self else { return }
+            
+            var snapshot = self.dataSource.snapshot()
+            
+            var contentItems: [EmotionsCollectionViewDataSource.ItemType] = self.selectedEmotions.compactMap({ emotion in
+                return .emotion(EmotionContentModel(emotion: emotion))
+            })
+            
+            if contentItems.isEmpty {
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Emotions/EmotionsViewController.swift]: Ensures we at least have 1 item at the top

File: iOS/Flows/Conversation/Emotions/EmotionsViewController.swift [hunk 5]
```diff
@@ -85,35 +76,94 @@ class EmotionsViewController: DiffableCollectionViewController<EmotionCategory,
...
+            snapshot.setItems(categoryItems, in: .categories)
+            await self.dataSource.apply(snapshot)
+        }
+    }
+    
+    private func handleSelected(emotion: Emotion) {
+        if self.selectedEmotions.contains(emotion) {
+            self.selectedEmotions.remove(object: emotion)
+        } else {
+            self.selectedEmotions.insert(emotion, at: 0)
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Emotions/EmotionsViewController.swift:88]: Here is where we insert at the beginning to ensure the most recent selection is visible.

**Review by MartinJibber — APPROVED** [review]

**Pull request merged**